src/txn/txn_rollback_to_stable.c

```diff
@@ -1000,74 +968,43 @@ static int
 __rollback_to_stable_btree_hs_truncate(WT_SESSION_IMPL *session, uint32_t btree_id)
 {
     WT_CURSOR *hs_cursor;
-    WT_CURSOR_BTREE *cbt;
     WT_DECL_ITEM(hs_key);
     WT_DECL_RET;
-    WT_ITEM key;
-    WT_UPDATE *hs_upd;
     wt_timestamp_t hs_start_ts;
     uint64_t hs_counter;
     uint32_t hs_btree_id;
-    int exact;
     char ts_string[WT_TS_INT_STRING_SIZE];
 
     hs_cursor = NULL;
-    WT_CLEAR(key);
-    hs_upd = NULL;
 
     WT_RET(__wt_scr_alloc(session, 0, &hs_key));
 
     /* Open a history store table cursor. */
-    WT_ERR(__wt_hs_cursor_open(session));
-    hs_cursor = session->hs_cursor;
-    cbt = (WT_CURSOR_BTREE *)hs_cursor;
+    WT_ERR(__wt_curhs_open(session, NULL, &hs_cursor));
 
     /* Walk the history store for the given btree. */
-    hs_cursor->set_key(hs_cursor, btree_id, &key, WT_TS_NONE, 0);
-    ret = __wt_hs_cursor_search_near(session, hs_cursor, &exact);
-
-    /*
-     * The search should always end up pointing to the start of the required btree or end of the
-     * previous btree on success. Move the cursor based on the result.
-     */
-    WT_ASSERT(session, (ret != 0 || exact != 0));
-    if (ret == 0 && exact < 0)
-        ret = __wt_hs_cursor_next(session, hs_cursor);
+    hs_cursor->set_key(hs_cursor, 1, btree_id);
+    ret = __wt_hs_cursor_search_near_after(session, hs_cursor);
```

Comment:
Why there is no handling for WT_NOTFOUND ?

Reply:
The for loop after this statement will not execute in case error is set. The line after for loop is handling the error.